### DISCUSSION TITLE
Timeouts in Vitest fast-check exceeded

### DISCUSSION THREAD

**Discussion by jasikpark** [dubzzz/fast-check#4109] [Q&A]
Does the vitest integration register each prop test run as a separate test?

That seems like it would be useful for allowing all of the tests to run in a test run, rather than timing out, ex:

```
✓ src/screens/hosts/validations/makeOverlayIPValidation.test.ts  (6 tests) 1192ms
 ❯ src/hooks/useCookie.test.tsx  (1 test | 1 failed) 5005ms
   ❯ src/hooks/useCookie.test.tsx > useCookie > should use, set, and delete a cookie with a provided key (with seed=-51)
     → Test timed out in 5000ms.
If this is a long-running test, pass a timeout value as the last argument or configure it globally with "testTimeout".
 ✓ src/screens/roles/detail/firewall-rule-form/ports-field/validatePorts.test.ts  (4 tests) 8ms
 ✓ src/util/cookies.test.ts  (5 tests) 76ms
```

Am I just holding it wrong? Is a faster test / a larger timeout manually added to the test the answer here?

**Comment by dubzzz** [comment]
So far, fast-check does not impact the timeout of test runners nor read it (for the later, we read it in the case of `@fast-check/jest` and stop fast-check runs in time)

**Reply by dubzzz** [reply]
I plan to have the same as Jest behavior for Vitest but I have not connected it for now. But first I'll need to check timeout on sync code.

**Reply by IanVS** [reply]
Can we use something like `interruptAfterTimeLimit` to control the amount of time that the check executes, to prevent it from hitting the timeout?

**Reply by dubzzz** [reply]
Yes, that's basically what does the package `@fast-check/jest`, with Jest

**Reply by IanVS** [reply]
OK, but just to confirm, there's no way that I can set that value manually when using it with vitest (`@fast-check/vitest`)?

**Reply by dubzzz** [reply]
You may do that via:

```ts
fc.configureGlobal({interruptAfterTimeLimit: 100})
```

Or via:

```ts
it.prop([...], {interruptAfterTimeLimit: 100})('...', (...) => {
  //...
})
```

**Comment by dubzzz** [comment]
I'm working onto a deeper integration with vitest. For now, I'm waiting for a fix on their side but as soon as I will have it I will be able to come up with a close integration with it and possibly better handle your case.

**Updated:** I just saw the fix has already been merged on their end. I missed the notification but now I've seen it I'll tackle the point.

**Updated<sup>2</sup>:** The fix I'll need only appears in [v1.0.0-beta.0](https://github.com/vitest-dev/vitest/releases/tag/v1.0.0-beta.0) for now. I'm experimenting with it.
